Show breakdown of blank line for total contribution in code panel #427

apoorva17 opened the pull request:
Fixes #427 
```
With the addition of #385, code view now shows a breakdown of total
contribution by filetype.

However, this does not provide any information on the blank LOC and
non-blank LOC.

Let's show a breakdown of total contribution by blank lines and
non-blank lines in the code view, to provide a better understanding of
the actual amount of code contributed by the author.

Let's also show this breakdown for each file type in the form of a
tooltip.
```
![dashboard](https://user-images.githubusercontent.com/19278550/50054044-a48e4000-0178-11e9-90da-506e0b782774.png)

apoorva17 commented:
@reposense/devs ready for review

yamidark commented on frontend/src/index.jade:
```diff
@@ -126,11 +126,12 @@ html
             span.fileType(v-if="totalLineCount > 0") &rarr; &nbsp;
                 template(v-for="type in Object.keys(filesLinesObj)")
-                    span {{ type }}: {{ filesLinesObj[type] }}
+                    span(v-bind:title="getBlankLineCount(type)") {{ type }}: {{ filesLinesObj[type] }}
```
Cant seem to see this portion on the netlify deploy?

apoorva17 replied:
it appears when you hover over the filetype in the code view in netlify

yamidark commented on frontend/src/static/js/v_authorship.js:
```diff
@@ -129,6 +146,11 @@ window.vAuthorship = {
           }), {});
     },
+
+    getBlankLineCount(fileType) {
```
This method name makes me think I'm just getting the blank line of the entire analysis counts as an `int`, but it returns a `string` instead with info on non-blank line count as well. Perhaps rename to `getFileBlankLineInfo`?

apoorva17 replied:
ok, have renamed the method

yamidark commented on frontend/src/static/js/v_authorship.js:
```diff
@@ -97,16 +110,20 @@ window.vAuthorship = {
 
-          const segments = this.splitSegments(file.lines);
-          out.segments = segments;
+          var segmentInfo = this.splitSegments(file.lines);
```
Any reason why this is changed to `var`?

apoorva17 replied:
have changed it back to ```const```

damithc commented:
> @damithc your opinion on this?

Sounds good.

apoorva17 commented:
@eugenepeh @damithc so u would like to retain the current tooltip display which shows the breakdown of blank and non-blank line for each filetype, and also add in the blank lines in brackets like below? but in such a case how would the user know what the value in the brackets represent?

> Total LoC contributed: 2316 (401)
> →  gradle: 47 (8), java: 2143 (363), js: 4 (0), md: 120 (30), yml: 2 (0) 
> ```
> where the number in the bracket denotes the blank lines.

eugenepeh commented:
> @eugenepeh @damithc so u would like to retain the current tooltip display which shows the breakdown of blank and non-blank line for each filetype, and add in the blank lines in brackets in the code view like below right? but would it still be confusing for the user as to what the values in the brackets represent?

Since it should be colored in gray, its not important information. Users who would like to know what that means can either hover over it or refer to the user guide for more info. So it shouldn't be much of a problem.

apoorva17 commented:
> Since it should be colored in gray, its not important information. Users who would like to know what that means can either hover over it or refer to the user guide for more info. So it shouldn't be much of a problem.

Ok, i've added in the values in brackets accordingly, and i've also added a tooltip for total LOC blank and non-blank lines breakdown.

yong24s commented:
Some comments on the UI, else LGTM. 👍 

Is it more user-friendly if the tooltip appears when hover for the entire `adoc: 304 (71)` rather than just the bracket `(71)`?

Also for the tooltip content, is it more intuitive for the `Non-Blank` to be on the left since it of interest to the user? `i.e.`  `adoc: Non-Blank: 233, Blank: 71`

Let's hear from @damithc first. 😄

damithc commented:
@apoorva17 can post a screenshot of the latest UI?

apoorva17 commented:
![dashboard](https://user-images.githubusercontent.com/19278550/50224769-aefb4480-03d9-11e9-9f88-50530151ef86.png)
The tooltip currently appears when user hovers over bracket value ```(1)```
> @apoorva17 can post a screenshot of the latest UI?

damithc commented:
I'm OK with the current UI.

> Is it more user-friendly if the tooltip appears when hover for the entire `adoc: 304 (71)` rather than just the bracket `(71)`?

Yes, that can be useful.

apoorva17 commented:
I've updated the implementation with this
> Is it more user-friendly if the tooltip appears when hover for the entire `adoc: 304 (71)` rather than just the bracket `(71)`?
